File: common/src/test/java/io/muun/common/utils/HexUtilsTest.java
package io.muun.common.utils;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class HexUtilsTest {

    @Test
    public void testEmptyEncode() throws Exception {
        final byte[] emptyArray = {};
        assertThat(Encodings.bytesToHex(emptyArray)).isEmpty();
    }

    @Test
    public void testValidEncode() throws Exception {
        final byte[] bytes = {0x01, 0x02, (byte) 0xFF};
        assertThat(Encodings.bytesToHex(bytes)).isEqualToIgnoringCase("0102ff");
    }

    @Test(expected = IllegalArgumentException.class)
    public void testDecodeOddLengthStringFails() throws Exception {
        Encodings.hexToBytes("a");
    }

    @Test(expected = IllegalArgumentException.class)
    public void testInvalidCharDecode() throws Exception {
        Encodings.hexToBytes("g");
    }

    @Test
    public void testDecodeEmptyString() throws Exception {
        assertThat(Encodings.hexToBytes("")).isEmpty();
    }

    @Test
    public void testValidUpperCaseDecode() throws Exception {
        final byte[] bytes = {0x01, 0x02, (byte) 0xFF};
        assertThat(Encodings.hexToBytes("0102FF")).isEqualTo(bytes);
    }

    @Test
    public void testValidLowerCaseDecode() throws Exception {
        final byte[] bytes = {(byte) 0xca, (byte) 0xb2, (byte) 0xFF};
        assertThat(Encodings.hexToBytes("cab2ff")).isEqualTo(bytes);
    }

    @Test
    public void testValidMixedCaseDecode() throws Exception {
        final byte[] bytes = {(byte) 0xaa, (byte) 0xbb, (byte) 0xdd};
        assertThat(Encodings.hexToBytes("AaBbdD")).isEqualTo(bytes);
    }

}

File: common/src/test/java/io/muun/common/utils/HexUtilsTest.kt
package io.muun.common.utils

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class HexUtilsTest {

    @Test
    @Throws(Exception::class)
    fun `test empty encode`() {
        val emptyArray = byteArrayOf()
        assertThat(Encodings.bytesToHex(emptyArray)).isEmpty()
    }

    @Test
    @Throws(Exception::class)
    fun `test valid encode`() {
        val bytes = byteArrayOf(0x01, 0x02, 0xFF.toByte())
        assertThat(Encodings.bytesToHex(bytes)).isEqualToIgnoringCase("0102ff")
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun `test decode odd length string fails`() {
        Encodings.hexToBytes("a")
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun `test invalid char decode`() {
        Encodings.hexToBytes("g")
    }

    @Test
    @Throws(Exception::class)
    fun `test decode empty string`() {
        assertThat(Encodings.hexToBytes("")).isEmpty()
    }

    @Test
    @Throws(Exception::class)
    fun `test valid upper case decode`() {
        val bytes = byteArrayOf(0x01, 0x02, 0xFF.toByte())
        assertThat(Encodings.hexToBytes("0102FF")).isEqualTo(bytes)
    }

    @Test
    @Throws(Exception::class)
    fun `test valid lower case decode`() {
        val bytes = byteArrayOf(0xca.toByte(), 0xb2.toByte(), 0xFF.toByte())
        assertThat(Encodings.hexToBytes("cab2ff")).isEqualTo(bytes)
    }

    @Test
    @Throws(Exception::class)
    fun `test valid mixed case decode 1`() {
        val bytes = byteArrayOf(0xaa.toByte(), 0xbb.toByte(), 0xdd.toByte())
        assertThat(Encodings.hexToBytes("AaBbdD")).isEqualTo(bytes)
    }

}